Can't build with latest code because of Java error

<!-- NOTE: If you ignore this template, we will send it again and ask you to fill it out anyway. -->

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
yes

**What version of Shaka Player are you using?**
NA main branch commit c3ff8e5e5f6ad5867ed0650e153627dafcb1bcf7

**Can you reproduce the issue with our latest release version?**


**Can you reproduce the issue with the latest code from `main`?**
Yes

**Are you using the demo app or your own custom app?**
NA

**If custom app, can you reproduce the issue using our demo app?**
NA

**What browser and OS are you using?**
NA

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**


**What are the manifest and license server URIs?**
<!-- NOTE:
  You can send the URIs to <shaka-player-issues@google.com> instead,
  but please use GitHub and the template for the rest.
  A copy of the manifest text or an attached manifest will **not** be
  enough to reproduce your issue, and we **will** ask you to send a
  URI instead.  You can copy the URI of the demo app to send us the
  exact asset, licence server, and settings you have selected there.
-->
NA

**What configuration are you using?  What is the output of `player.getConfiguration()`?**
<!-- NOTE:
  You can censor URLs to keep them private, but include them in the email.  You
  can also use JSON.stringify(obj, null, 2) to print nicely on platforms that
  don't print objects well in the console.  DON'T SEND '[object Object]'!
-->
NA

**What did you do?**
<!-- Steps to reproduce the bug -->
Tried to build with the command python build/all.py

**What did you expect to happen?**


**What actually happened?**
Build failed with this error message

P2134253@CHTRMACP4H2J3 shaka-player % python build/all.py  
[WARNING] No changes detected, skipping. Use --force to override.
[INFO] Generating Closure dependencies...
[INFO] Linting JavaScript...
[WARNING] No changes detected, skipping. Use --force to override.
[INFO] Linting CSS...
[WARNING] No changes detected, skipping. Use --force to override.
[INFO] Linting HTML...
[WARNING] No changes detected, skipping. Use --force to override.
[INFO] Checking that the build files are complete...
[INFO] Checking for common misspellings...
[INFO] Checking correct usage of eslint-disable...
[INFO] Checking the tests for type errors...
[WARNING] No changes detected, skipping. Use --force to override.
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: com/google/javascript/jscomp/CommandLineRunner has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:601)
[ERROR] Build failed
P2134253@CHTRMACP4H2J3 shaka-player % java -version
**java version "1.8.0_351"**
Java(TM) SE Runtime Environment (build 1.8.0_351-b10)
Java HotSpot(TM) 64-Bit Server VM (build 25.351-b10, mixed mode)
P2134253@CHTRMACP4H2J3 shaka-player % node --version
v16.13.1
P2134253@CHTRMACP4H2J3 shaka-player % python --version
Python 3.10.5

**I was able to work around the problem by reverting changes to the closure versions in package.json** 

```
-    "google-closure-compiler-java": "^20220301.0.0",
-    "google-closure-deps": "https://gitpkg.now.sh/google/closure-library/closure-deps?d7736da6",
-    "google-closure-library": "^20220301.0.0",
+    "google-closure-compiler-java": "^20221102.0.0",
+    "google-closure-deps": "^20221102.0.0",
+    "google-closure-library": "^20221102.0.0",

```
**from this commit**

fix(hls): Fix type error in lazy-loading (#4687)

There was a potential type error in an edge case in lib/player.js. It 
would have been caught by the latest compiler, so this upgrades the 
compiler and fixes another type error in one other place, as well.
Commit:
28b73b921d1dbdc6d7e016aa8e0a000e62318ed3 [28b73b92]
Parents:
7af357cf7cd211791c25c69b46745c896fb1c827
Author:
Joey Parrish <joeyparrish@users.noreply.github.com>
Date:
November 11, 2022 at 8:07:58 AM MST
Committer:
GitHub <noreply@github.com>




**This problem was observed on an M1 macbook running MacOS version 12.6**

I am running the latest available Java version for MacOS

--- a/build/install-linux-prereqs.sh
+++ b/build/install-linux-prereqs.sh
@@ -2,15 +2,14 @@
 
 # Installs prerequisites for Shaka Player development on Linux:
 #  - Git v1.9+ (to check out the code)
-#  - Python v2.7 (to run the build scripts)
-#  - Java Runtime Environment v8+ (to run the Closure compiler)
+#  - Python v2.7 or v3.5+ (to run the build scripts)
+#  - Java Runtime Environment v11+ (to run the Closure compiler)
 #  - Apache (to serve the Shaka Player demo)
-#  - NodeJS v8+ (to run some build deps, such as jsdoc, karma, etc)
+#  - NodeJS v14+ (to run some build deps, such as jsdoc, karma, etc)
 
 # Tested on:
-#  - Ubuntu 16.04 LTS (Xenial)
-#  - Ubuntu 17.10 (Artful)
-#  - Debian 9 (Stretch)
+#  - Ubuntu 18.04 LTS (Bionic)
+#  - Debian 10 (Buster)
 #  - gLinux
 
 # Tested with:
@@ -19,8 +18,10 @@
 #  - Previous install/upgrade of nodejs/npm through n or nvm
 
 # Not supported:
+#  - Ubuntu 16.04 LTS (Xenial) - Java 9
 #  - Ubuntu 14.04 LTS (Trusty) - Java 7
 #  - Debian 8 (Jessie) - Java 7
+#  - Debian 9 (Stretch) - Java 8
 #  - RHEL/CentOS 6 - python 2.6, git 1.7
 #  - RHEL/CentOS 7 - git 1.8
 
@@ -53,14 +54,14 @@ echo "*****" 1>&2
 echo "Updating packages and installing git, Python, JRE, and Apache." 1>&2
 echo "*****" 1>&2
 sudo apt -y update
-sudo apt -y install git python2.7 default-jre-headless apache2
+sudo apt -y install git python default-jre-headless apache2
 
 # NodeJS in Ubuntu and Debian is often out of date, so we may need to grab a
-# newer version.  We require v8+.
+# newer version.  We require v14+.
 
-if node --version 2>/dev/null | grep -q 'v\([89]\|1[0-9]\)'; then
+if node --version 2>/dev/null | grep -q 'v\(1[4-9]\|[2-9][0-9]\)'; then
   echo "*****" 1>&2
-  echo "NodeJS v8+ detected.  No update needed." 1>&2
+  echo "NodeJS v14+ detected.  No update needed." 1>&2
   echo "*****" 1>&2
 else
   if node --version &>/dev/null; then
@@ -101,7 +102,7 @@ else
       fi
     else
       echo "*****" 1>&2
-      echo "You will need to upgrade NodeJS yourself to v8+." 1>&2
+      echo "You will need to upgrade NodeJS yourself to v14+." 1>&2
       echo "*****" 1>&2
       exit 1
     fi
@@ -115,22 +116,22 @@ else
       | grep '^Version:' | cut -f 2 -d ' ')
   available_node_major_version=$(echo "$available_node_version" \
       | cut -f 1 -d '.')
-  if [ "$available_node_major_version" -gt "8" ]; then
+  if [ "$available_node_major_version" -gt "13" ]; then
     sudo apt -y install nodejs npm
     echo "*****" 1>&2
     echo "NodeJS v$available_node_version installed from your distro." 1>&2
     echo "*****" 1>&2
   else
     # Fetch a known-good copy of NodeJS from nodesource.com and install it.
-    # NodeJS v10.19.0 has npm v6.13.4.
+    # NodeJS v14.21.1 has npm v6.14.17.
     deb_file=$(mktemp --suffix .deb)
     curl -o "$deb_file" \
-        https://deb.nodesource.com/node_10.x/pool/main/n/nodejs/nodejs_10.19.0-1nodesource1_amd64.deb
+        https://deb.nodesource.com/node_14.x/pool/main/n/nodejs/nodejs_14.21.1-1nodesource1_amd64.deb
     sudo dpkg -i "$deb_file"
     rm -f "$deb_file"
 
     echo "*****" 1>&2
-    echo "NodeJS v10.19 installed from nodesource." 1>&2
+    echo "NodeJS v14.21 installed from nodesource." 1>&2
     echo "*****" 1>&2
   fi
 fi

--- a/docs/tutorials/welcome.md
+++ b/docs/tutorials/welcome.md
@@ -29,8 +29,8 @@ You can build Shaka on Linux, Windows, or Mac.
 To get the sources and compile the library, you will need:
   * {@link https://git-scm.com/downloads Git v1.9+}
   * {@link https://www.python.org/downloads/ Python v2.7 or v3.5+}
-  * {@link https://java.com/en/download/ Java Runtime Environment v8+}
-  * {@link https://nodejs.org/en/download/ NodeJS v10+}
+  * {@link https://java.com/en/download/ Java Runtime Environment v14+}
+  * {@link https://nodejs.org/en/download/ NodeJS v14+}
   * A local web server, such as {@link https://httpd.apache.org/ Apache}
     * _NOTE: A local web server is required because browsers place restrictions
       on applications from file:/// URLs._

